examples/todos/src/containers/VisibleTodoList.spec.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { mount, shallow } from 'enzyme'
+import { createStore } from 'redux'
+import { Provider } from 'react-redux'
+
+import rootReducer from '../reducers'
+import VisibleTodoList, { mapStateToProps } from './VisibleTodoList';
+import TodoList from '../Components/TodoList';
+import { VisibilityFilters } from '../actions'
+
+
+
+describe("integration test, high verification", () => {
+    it('renders an empty list', () => {
+        const store = createStore(rootReducer)
+        const wrapper = mount(
+            <Provider store={store}>
+                <VisibleTodoList />
+            </Provider>
+        )
+        expect(wrapper.html()).toBe('<ul></ul>')
+    })
+})
+
+describe("unit tests, high isolation", () => {
+    describe("mapStateToProps", () => {
+        it('maps an empty list to empty list', () => {
+            const props = mapStateToProps({
+                todos: [],
+                visibilityFilter: VisibilityFilters.SHOW_ALL
+            })
+            expect(props).toEqual({ todos: [] })
+        });
+    })
+
+    describe("<VisibleTodoList />", () => {
+        it('supplies an empty list to the <TodoList />', () => {
+            const store = createStore(rootReducer)
+            const wrapper = mount(
+                <Provider store={store}>
+                    <VisibleTodoList />
+                </Provider>
+            )
+            expect(wrapper.find('TodoList').prop('todos')).toEqual([])
+        })
+    });
+
+    describe("<TodoList />", () => {
+        it('renders an empty list', () => {
+            const wrapper = shallow(
```

Comment:
@kentcdodds I believe RTL also does not have this feature to mock out all children (I'm aware Enzyme does more than exactly this). You'd have to manually maintain explicit lists of mocks as they change over time for each test that is written in this style, in RTL?

Reply:
https://kcd.im/shallow

Reply:
I've tried to engage you in a debate on this before, and only got the URL. I'm well aware of your position on shallow :)  Objectively speaking though, to write a test in this style in RTL requires manually maintaining a list of children to mock, correct? It has no auto-mock children feature like Angular or Enzyme, right?

Reply:
I've given links to my blog posts to literally thousands of people, so sorry that I don't remember engaging in this debate with you. That said, since you already know my opinion on it I'm not sure why you were asking...

The fact is, I don't write tests "in this style" and I think they're more harmful than helpful.

Reply:
No worries :) Thanks for replying.

I wasn't asking you if you think devs should *should* write these tests in general, I was asking if RTL has affordances to do a particular thing, which is an objective question not a matter of opinion:

> You'd have to manually maintain explicit lists of mocks [of children] as they change over time for each test that is written in this style, in RTL?

Sorry if that was not clear, or any my comments came off as antagonizing in any way. I don't intend for my abrasive communication style to get in the way of the point I'm trying to make here, so sorry if it came off that way :)

>  I think they're more harmful than helpful.

Generally speaking, I would agree with you & I would recommend people avoid "this style" (asserting on implementation details) as well. However, having no tests can be even more harmful than having less than ideal tests written tactfully, so asserting on implementation details can be the lesser of two evils if you're in that pickle. I really like this article [https://codingitwrong.com/2018/12/03/why-you-should-sometimes-test-implementation-details.html](https://codingitwrong.com/2018/12/03/why-you-should-sometimes-test-implementation-details.html) and would be interested in your thoughts after giving it a good read, if you were so inclined.

My position that is people can make a mess of their test suites with any library (although its much easier to make a mess with Enzyme than RTL). But if there is a mess of hard to test code that is entrenched in your project, for whatever reason, Enzyme's escape hatches can help a more experienced dev come in & "fix things" & can be a life saver... I agree Enzyme can give devs a huge foot gun with no warning on the tin. But dangerous tools can be useful, too...